lib/recurly/hosted-field.js

```diff
@@ -85,20 +87,33 @@ export class HostedField extends Emitter {
   }
 
   inject () {
-    this.target.innerHTML = `
+    const hostedDiv = `
       <div class="${this.classList}">
-        <iframe
-          src="${this.url}"
-          border="0"
-          frameborder="0"
-          allowtransparency="true"
-          scrolling="no">
-        </iframe>
       </div>
     `;
 
-    this.container = this.target.children[0];
-    this.iframe = this.container.children[0];
+    const hostedIframe = `
+      <iframe
+        src="${this.url}"
+        border="0"
+        frameborder="0"
+        allowtransparency="true"
+        scrolling="no">
+      </iframe>
+    `;
+
+    this.hiddenInputProxy = dom.createHiddenInput();
+    this.hiddenInputProxy.addEventListener('focus', () => {
+      this.focus();
+    });
+
+    this.target.innerHTML = hostedDiv;
+
+    this.target.children[0].insertAdjacentElement('afterbegin', this.hiddenInputProxy);
+    this.target.children[0].children[0].insertAdjacentHTML('afterend', hostedIframe);
```

Comment:
Instead of building and inserting this separately from the html text above, let's just write it into the `this.target.innerHTML` assignment, then find it in the injected DOM structure as we do for `this.container`, `this.iframe`, etc. As this is written, it's difficult to tell where the input proxy is meant to reside in the tree.